### PR TITLE
feat: Use non root user as default user in container image

### PR DESCRIPTION
**Pull request by timo-42**
Signed-off-by: Timo Haas <haastimo@gmx.de>

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Changed default user in container images from 0 to 1001

## Verification

Tested thanos-query in k8s cluster

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -5,4 +5,5 @@ LABEL maintainer="The Thanos Authors"
 
 COPY /thanos_tmp_for_docker /bin/thanos
 
+USER 1001
```
Comment (matej-g) [inline Dockerfile:8]: Thanks for adding this! Just confirming, we don't need to first do `useradd`?

Comment (timo-42) [reply Dockerfile:14]: @matej-g I added your suggestion.

**Review by matej-g — APPROVED** [review]

**Comment by matej-g** [comment]
Thank you @timo-42!

**Comment by WyrickC** [comment]
Hi, just wondering why this change was made? We have our thanos compactor running as any UID and this change is breaking for us. We can get it to work in our K8s cluster by making sure it runs as root, but we'd rather not have it do that. Any suggestions?

**Pull request merged**